### ISSUE TITLE
[Feature Request] Being able to add slint into a custom wgpu render like egui

### ISSUE THREAD

**Issue by ValorZard** [slint-ui/slint#4499]
For other rust UI llibraries like [egui](https://github.com/emilk/egui) and [iced](https://github.com/iced-rs/iced)), they have great integration with wgpu, making so that its relatively easy to integrate it with other projects that use wgpu like the Bevy Game Engine. (See [bevy_egui](https://github.com/mvlabat/bevy_egui) and [bevy_iced](https://github.com/tasgon/bevy_iced) It would be neat if Sllint could do the same thing!

Inspired by this reddit comment: https://www.reddit.com/r/rust/comments/1afhcvz/slint_v140_declarative_gui_for_rust/koap3lv/

**Comment by tronical** [comment]
Fully agreed. This would be a nice demo. In theory it's already possible, if selecting the gles wgpu backend and combining with the femtovg renderer.

Mid term it would be nice to decouple them, render with slint into a backend specific texture, convert to an OS specific buffer handle (dmabuf, iosurface, ...) and then import into wgpu. Comes at the expense of additional passes but if the UI doesn't change too much it might be just fine.

**Comment by AdrianEddy** [comment]
+1. 
Having eg. [vello](https://github.com/linebender/vello) as a renderer would be amazing, it's getting close to a first release and I wonder how would the performance compare to current renderers.
Also for me personally it's important for the UI to run on each platform's native backend, so if Slint would run on DirectX, Metal, OpenGL and Vulkan, instead of just OpenGL, that would be great (I know skia can run on DirectX and Metal, but wgpu would be better suited for my use case and it adds Vulkan)

**Comment by tronical** [comment]
I agree, vello would be great. I hope they will make a release to crates.io.

**Comment by WilstonOreo** [comment]
@tronical  @AdrianEddy  https://github.com/linebender/vello has released a 0.1.0 two days ago

**Comment by tronical** [comment]
Meanwhile, another crazy idea I've had is to initialise the underlying graphics APIs (Metal, Vulkan, D3D) with wgpu, then pass the `as_hal`/`as_raw` provided command queues, devices, etc. to Skia, render with Skia, and meanwhile provide the wgpu types to rendering notifiers and (most importantly) allow creating a `slint::Image` from a `wgpu::Texture`. If this works, we might be able to use wgpu as the more universal API and interface for custom rendering, while still being able to benefit from Skia. But I admit, it's a bit crazy and might not work :)

**Comment by tronical** [comment]
In principle this is implemented now and will be in the upcoming 1.12 release:

https://snapshots.slint.dev/master/docs/rust/slint/wgpu_24/

https://github.com/slint-ui/slint/tree/master/examples/bevy

There is more work to be done (see skia comment, also vello renderer), but the building blocks are in place.
